### PR TITLE
Implement the happy path for creating audiences and the custom dimension via the Setup CTA Banner.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

Addresses issue:

- #8131

## Relevant technical choices

As discussed in a recent standup, I realised during implementation that reusing the partial data infrastructure as described in the IB is not such a good fit here, so instead I've taken the approach of retrieving a regular Analytics report for the 90 day range to determine whether data is available. This does mean we can't take advantage of a cached data availability state which is available on page load in order to avoid layout shifts when showing the banner; I will create a followup issue to investigate further options on that front.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/audiences.js [hunk 1]
```diff
@@ -127,8 +127,6 @@ const baseResolvers = {
 		if ( audiences === null ) {
 			yield fetchSyncAvailableAudiencesStore.actions.fetchSyncAvailableAudiences();
 		}
-
-		return registry.select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
```
Comment (techanvil) [inline assets/js/modules/analytics-4/datastore/audiences.js:131]: This addresses a minor unrelated issue flagged here: https://github.com/google/site-kit-wp/pull/8606#discussion_r1589434092

File: assets/js/modules/analytics-4/datastore/audiences.test.js [hunk 2]
```diff
@@ -177,11 +177,6 @@ describe( 'modules/analytics-4 audiences', () => {
 			];
 
 			it( 'should not sync cached audiences when the availableAudiences setting is not null', () => {
-				fetchMock.postOnce( syncAvailableAudiencesEndpoint, {
```
Comment (techanvil) [inline assets/js/modules/analytics-4/datastore/audiences.test.js:180]: This is another minor unrelated fix, see https://github.com/google/site-kit-wp/pull/8606#discussion_r1589436256.

File: assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss [hunk 3]
```diff
@@ -36,7 +36,7 @@
 		display: none;
 
 		@media (min-width: $bp-tablet) {
-			display: block;
+			display: flex;
```
Comment (techanvil) [inline assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss:39]: Fix the icon height and thus vertical alignment.

File: assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss [hunk 4]
```diff
@@ -60,7 +60,10 @@
 	}
 
 	.mdc-button {
+		font-weight: $fw-medium;
 		margin: 0 auto;
+		min-height: 32px;
```
Comment (techanvil) [inline assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss:65]: Noting that the primary CTA button is the shorter version, it follows the secondary "Got it" CTA should also be, therefore this stylign can be applied across the board to `.mdc_button` within the component (hence also the VRT update, which seems fine as we didn't have a mobile design for reference in the first place).

File: assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss [hunk 5]
```diff
@@ -60,7 +60,10 @@
...
+		padding: 6px 16px;
```
Comment (techanvil) [inline assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss:66]: Noting that the primary CTA button is the shorter version, it follows the secondary "Got it" CTA should also be, therefore this styling can be applied across the board to `.mdc_button` within the component (hence also the VRT update, which seems fine as we didn't have a mobile design for reference in the first place, and there's no meaningful difference to the wider viewport versions.).

Comment (techanvil) [inline assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss:66]: Noting that the primary CTA button is the shorter version, it follows the tertiary "Got it" CTA should also be, therefore this styling can be applied across the board to `.mdc_button` within the component (hence also the VRT update, which seems fine as we didn't have a mobile design for reference in the first place, and there's no meaningful difference to the wider viewport versions).

Note that both buttons are also `font-weight: 500`, the "Got it" tertiary CTA already having this weight.

Comment (techanvil) [inline assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss:66]: Noting that the primary CTA button is the shorter version, it follows the tertiary "Got it" CTA should also be, therefore this styling can be applied across the board to `.mdc_button` within the component (hence also the VRT update, which seems fine as we didn't have a mobile design for reference in the first place (see [Figma](https://www.figma.com/design/7ba0pj1rLuvLvJhy3NiHOj/AdSense?node-id=100-2626&t=wcYZG4atKxLUT3pB-0), theres's no meaningful difference to the wider viewport versions, and the new bottom margin _is_ aligned with the Figma design for the mobile version of the [Audience Segmentation success banner](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=1711-17131&t=nBgPqJ7Y8YIAsiwR-0)).

Note that both buttons are also `font-weight: 500`, the "Got it" tertiary CTA already having this weight.

File: assets/js/modules/analytics-4/datastore/partial-data.test.js [hunk 6]
```diff
@@ -187,6 +195,673 @@ describe( 'modules/analytics-4 partial data', () => {
...
+							...availableReturningVisitorsAudienceFixture,
+							name: createdReturningVisitorsAudienceName,
+						},
+					],
+				];
+
+				fetchMock.post(
+					{
+						url: syncAvailableAudiencesEndpoint,
+						repeat: 2,
```
Comment (techanvil) [inline assets/js/modules/analytics-4/datastore/partial-data.test.js:615]: Only allow two mocked calls to this endpoint.

**Review by hussain-t — CHANGES_REQUESTED** [review]
Excellent work, @techanvil 🎖️ 

I've left a few minor comments. Could you take a look?

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupSuccessSubtleNotification.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+	function scrollToWidgetArea() {
+		dismissNotificationForUser();
+
+		// TODO: Scrolling to the widget area will be implemented in a subsequent issue.
+	}
+
+	// Ensure resolution of the report has completed before showing this
+	// notification, since it should only appear when the user has no data in
+	// the report.
```
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupSuccessSubtleNotification.js:75]: Thanks @hussain-t, good spot. This comment isn't relevant here, and I've removed it.

File: assets/js/modules/analytics-4/datastore/audiences.test.js [hunk 8]
```diff
@@ -300,6 +301,684 @@ describe( 'modules/analytics-4 audiences', () => {
...
+
+			function createAvailableUserAudience( audienceID ) {
+				return {
+					name: `properties/12345/audiences/${ audienceID }`,
+					description: `Description ${ audienceID }`,
+					displayName: `Test Audience ${ audienceID }`,
+					audienceType: 'USER_AUDIENCE',
+					audienceSlug: '',
+				};
+			}
```
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/modules/analytics-4/datastore/audiences.test.js:338]: Thanks @hussain-t, another good spot - I've moved this and its sibling functions above the variable definitions.

File: assets/js/modules/analytics-4/datastore/audiences.test.js [hunk 9]
```diff
@@ -300,6 +301,684 @@ describe( 'modules/analytics-4 audiences', () => {
...
+						audienceResourceName: audiences.map(
+							( { name } ) => name
+						),
+					},
+					startDate,
+					endDate: referenceDate,
+				};
+			}
+
+			function createMockReport( totalUsersByAudience ) {
```
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/modules/analytics-4/datastore/audiences.test.js:354]: It's not clear we'll actually need it outside of these tests, although it's certainly possible.

I would suggest keeping it where it is for now and only extract it if we need to. I have in the meantime renamed it to be a bit more explicit.

File: assets/js/modules/analytics-4/datastore/audiences.test.js [hunk 10]
```diff
@@ -300,6 +301,684 @@ describe( 'modules/analytics-4 audiences', () => {
...
+			describe( 'custom dimension handling', () => {
+				const createCustomDimensionEndpoint = new RegExp(
+					'^/google-site-kit/v1/modules/analytics-4/data/create-custom-dimension'
+				);
+				const syncAvailableCustomDimensionsEndpoint = new RegExp(
+					'^/google-site-kit/v1/modules/analytics-4/data/sync-custom-dimensions'
+				);
+
+				const postTypeCustomDimension = {
+					parameterName: 'googlesitekit_post_type',
```
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/modules/analytics-4/datastore/audiences.test.js:876]: It's a fair shout to use the constant here - I've simply inlined it at the point of use.

File: assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (techanvil) [reply assets/sass/components/dashboard/_googlesitekit-subtle-notification.scss:66]: SGTM!

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 11]
```diff
@@ -42,93 +51,146 @@ import BannerGraphicsSVGDesktop from '../../../../../../svg/graphics/audience-se
...
+		const args = {
+			metrics: [ { name: 'totalUsers' } ],
+			startDate,
+			endDate,
+		};
+
+		return select( MODULES_ANALYTICS_4 ).hasZeroData( args ) === false;
+	} );
+
+	if ( configuredAudiences !== null || ! hasDataWithinPast90Days ) {
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:91]: Hey @hussain-t, just want to flag I made an additional change following your review which is to ensure the banner doesn't display while `configuredAudiences` is loading and therefore `undefined`. The change is in this commit: https://github.com/google/site-kit-wp/pull/8694/commits/51540e5ea89163f3df4b62b92d503fd937853ef6

Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:91]: Sounds good. Thanks, @techanvil!
